code/_core/mob/movement.dm

```diff
@@ -74,11 +74,11 @@
 		if(loc != old_loc)
 			post_move(old_loc)
 
-	if(length(observers))
-		for(var/mob/chosenObserver in observers)
-			chosenObserver.glide_size = glide_size
-			chosenObserver.Move(NewLoc, Dir, step_x, step_y, control_move = FALSE)
-	if(observed && control_move)
+	for(var/k in observers)
```

Comment:
Actually I'm an idiot and should've suggested all of this should go in ``mob/post_move(var/atom/old_loc)`` and other things but I can do that myself.